frontend/src/components/ArchiveVaultDialog.vue

```diff
@@ -0,0 +1,99 @@
+                          {{ t('archiveVaultDialog.description') }}
+                        </p>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+                <div class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">
+                  <button type="submit" class="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm" >
+                    {{ t('archiveVaultDialog.confirm') }}
+                  </button>
+                  <button type="button" class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm" @click="open = false">
+                    {{ t('common.cancel') }}
+                  </button>
+                </div>
+                <p v-if="onArchiveVaultError != null" class="text-sm text-red-900 px-4 sm:px-6 text-right bg-red-50">
+                  {{ t('common.unexpectedError', [onArchiveVaultError.message]) }}
+                </p>
+              </form>
+            </DialogPanel>
+          </TransitionChild>
+        </div>
+      </div>
+    </Dialog>
+  </TransitionRoot>
+</template>
+
+<script setup lang="ts">
+import { Dialog, DialogOverlay, DialogPanel, DialogTitle, TransitionChild, TransitionRoot } from '@headlessui/vue';
+import { ExclamationTriangleIcon } from '@heroicons/vue/24/outline';
+import { ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import backend, { VaultDto } from '../common/backend';
+
+const { t } = useI18n({ useScope: 'global' });
+
+const open = ref(false);
+
+const onArchiveVaultError = ref<Error | null>();
+
+const props = defineProps<{
+  vault: VaultDto
+}>();
+
+const emit = defineEmits<{
+  (e: 'close'): void
+  (e: 'archived', updatedVault: VaultDto): void
+}>();
+
+defineExpose({
+  show
+});
+
+function show() {
+  open.value = true;
+}
+
+async function archiveVault() {
+  onArchiveVaultError.value = null;
+  const v = props.vault;
+  try {
+    const updatedVault = await backend.vaults.createOrUpdateVault(v.id, v.name, v.description, true, v.masterkey, v.iterations, v.salt, v.authPublicKey, v.authPrivateKey );
+    if (updatedVault != null){
+      emit('archived', updatedVault);
+    } else {
+      //TODO: or throw error in this case? But the "damage" is already done in backend
+      const createdVault: VaultDto = { id: v.id, name: v.name, description: v.description, archived: true, creationTime: v.creationTime, masterkey: v.masterkey, iterations: v.iterations, salt: v.salt, authPublicKey: v.authPublicKey, authPrivateKey: v.authPrivateKey };
+      emit('archived', createdVault);
+    }
```

Comment:
I'm wondering about these lines of code (and also the same ones in `ReactivateVaultDialog`). Do we really want to make the distinction between "create" and "update" by a missing/existing response object? If yes, this should be documented in the backend API.

And, honestly, I think your TODO includes a legitimate question. What do we even do if a new vault has been created instead? In theory, we can ignore this case because vaults can't be deleted at the moment. But that doesn't have to be the case forever.

Reply:
>Do we really want to make the distinction between "create" and "update" by a missing/existing response object? 

Good question! I'll change the code to always return the object, will also clean up code. Creation/Update can still be differentiated with the response code.

> What do we even do if a new vault has been created instead? In theory, we can ignore this case because vaults can't be deleted at the moment. But that doesn't have to be the case forever.

Had the same thoughts. In my opinion, even for the long term this is okay, because deletion should really be the last option. If we ever allow deletion, we have to revise more than just archiving/reactivating anyway.